devolutions-gateway/src/http/middlewares/sogar_auth.rs

```diff
@@ -0,0 +1,105 @@
+use crate::{
+    config::{Config, SogarPermission, SogarUser},
+    http::middlewares::auth::{parse_auth_header, AuthHeaderType},
+};
+use picky::jose::jwt::{JwtSig, JwtValidator};
+use saphir::{http, http_context::State, prelude::*, response::Builder as ResponseBuilder};
+use slog_scope::error;
+use sogar_core::sogar_registry::{
+    BLOB_DOWNLOAD_ENDPOINT, BLOB_EXIST_ENDPOINT, BLOB_GET_LOCATION_ENDPOINT, BLOB_UPLOAD_ENDPOINT,
+    MANIFEST_DOWNLOAD_ENDPOINT, MANIFEST_EXIST_ENDPOINT, MANIFEST_UPLOAD_ENDPOINT,
+};
+use std::sync::Arc;
+
+pub struct SogarAuthMiddleware {
+    config: Arc<Config>,
+}
+
+impl SogarAuthMiddleware {
+    pub fn new(config: Arc<Config>) -> Self {
+        Self { config }
+    }
+}
+
+impl Middleware for SogarAuthMiddleware {
+    fn next(
+        &'static self,
+        ctx: HttpContext,
+        chain: &'static dyn MiddlewareChain,
+    ) -> BoxFuture<'static, Result<HttpContext, SaphirError>> {
+        auth_middleware(ctx, chain, self.config.clone()).boxed()
+    }
+}
+
+async fn auth_middleware(
+    ctx: HttpContext,
+    chain: &'static dyn MiddlewareChain,
+    config: Arc<Config>,
+) -> Result<HttpContext, SaphirError> {
+    if let Some(metadata) = ctx.metadata.name {
+        let auth_header = ctx
+            .state
+            .request()
+            .expect("Invalid middleware state")
+            .headers()
+            .get(http::header::AUTHORIZATION);
+
+        if auth_header.is_none() || auth_header.unwrap().to_str().is_err() {
+            error!("Authorization header is missing or wrong format.");
+            //to be able to play video in the browser
+            return if metadata == BLOB_DOWNLOAD_ENDPOINT || metadata == MANIFEST_DOWNLOAD_ENDPOINT {
```

Comment:
in other requests the auth header is checked to verify that user have access to this action. In order to be able to play video files in the browser, the header can be missed for the get requests. (but if header will be present it will be checked)